client/starwhale/api/_impl/wrapper.py

```diff
@@ -85,19 +85,27 @@ def __init__(self, eval_id: str, project: str, instance: str = ""):
         self.eval_id = eval_id
         self.project = project
         self.instance = instance
-        self._results_table_name = self._get_datastore_table_name("results")
-        self._summary_table_name = data_store.gen_table_name(
-            project=self.project, table="eval/summary", instance_uri=self.instance
+        self._tables: Dict[str, str] = {}
+        self._eval_table_name: Callable[
+            [str], str
+        ] = (
+            lambda name: f"eval/{self.eval_id[:VERSION_PREFIX_CNT]}/{self.eval_id}/{name}"
         )
+        self._eval_summary_table_name = "eval/summary"
         self._data_store = data_store.get_data_store(instance_uri=instance)
-        self._init_writers([self._results_table_name, self._summary_table_name])
+        self._init_writers([])
 
-    def _get_datastore_table_name(self, name: str) -> str:
-        return data_store.gen_table_name(
-            project=self.project,
-            table=f"eval/{self.eval_id[:VERSION_PREFIX_CNT]}/{self.eval_id}/{name}",
-            instance_uri=self.instance,
-        )
+    def _fetch_table_name(self, table: str) -> str:
+        with self._lock:
+            _table_name = self._tables.get(table)
+            if _table_name is None:
+                _table_name = data_store.gen_table_name(
+                    project=self.project,
+                    table=table,
+                    instance_uri=self.instance,
+                )
```

Comment:
It seems like there will be a network request for each table name getting.
Add a cache decorator for `_get_remote_project_id` ?

Reply:
done